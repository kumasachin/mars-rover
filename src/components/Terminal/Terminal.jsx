import React from 'react';
import "./Terminal.css";

export const Terminal = ({printLogs}) => {
    const logs = printLogs ?
        printLogs.map((print, index) => {
            return (
                print.currentCoordinate ? <div className={index === 0 ? "one-robot-latest-log" :  "one-robot-log" }>
                    >{print.name}<br />
                    >{print.currentCoordinate}<br />
                    >{print.instruction}<br /><br /><br />
                    >{print.isLost && "Lost"}<br /><br /><br />
                </div> : <div className={index === 0 ? "one-map-latest-log" :  "one-map-log" }>
                    >
                </div>
            )
        }): null;
    return (
        <>
             <div class="consolebody">
                <p>></p>
                {logs}
            </div>
        </>
    );
};

export default Terminal;