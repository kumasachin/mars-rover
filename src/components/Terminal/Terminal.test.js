import { render, screen, cleanup } from '@testing-library/react';
import Terminal from './Terminal';

describe("Trades", () => {
  afterEach(cleanup);

  it('should take a snapshot', () => {
    const { asFragment } = render(<Terminal />)
    expect(asFragment(<Terminal />)).toMatchSnapshot()
  });

});