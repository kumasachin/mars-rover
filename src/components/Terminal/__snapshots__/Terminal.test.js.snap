// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trades should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="consolebody"
  >
    <p>
      &gt;
    </p>
  </div>
</DocumentFragment>
`;
